File: php/test/Structural/FacadeTest.php
<?

require_once 'src/Structural/Facade/FacadeCuppaMaker.php';

class Structural_FacadeTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$cuppaMaker = new FacadeCuppaMaker();
		$teaCup = $cuppaMaker->makeACuppa();
		$this->assertEquals($teaCup->toString() ,'A nice cuppa tea!', 'Test tea cup facade');
	}
}

File: go/structural/facade/facade_test.go
//TestFacade - testing the Facade

package facade

import (
	"testing"
	"../../lib/asserts"
)

func TestFacade( t *testing.T ) {

	cuppaMaker := NewFacadeCuppaMaker()
	teaCup := cuppaMaker.MakeACuppa()
	asserts.Equals( t, "Facade test", 
		"A nice cuppa tea!",
		teaCup.ToString() )
}
